src/cool_seq_tool/mappers/exon_genomic_coords.py

```diff
@@ -368,67 +389,89 @@ async def genomic_to_tx_segment(
         resp.genomic_data = GenomicData(**params)
         return resp
 
-    @staticmethod
-    def _validate_exon(
-        transcript: str, tx_exons: list[tuple[int, int]], exon_number: int
-    ) -> tuple[tuple[int, int] | None, str | None]:
-        """Validate that exon number exists on a given transcript
+    async def _get_all_exon_coords(
+        self, tx_ac: str, genomic_ac: str | None = None
+    ) -> list[ExonCoord]:
+        """Get all exon coordinate data for a transcript.
 
-        :param transcript: Transcript accession
-        :param tx_exons: List of transcript's exons and associated coordinates
-        :param exon_number: Exon number to validate
-        :return: Exon coordinates for a given exon number and warnings if found
-        """
-        msg = f"Exon {exon_number} does not exist on {transcript}"
-        try:
-            if exon_number < 1:
-                return None, msg
-            exon = tx_exons[exon_number - 1]
-        except IndexError:
-            return None, msg
-        return exon, None
+        If ``genomic_ac`` is NOT provided, this method will use the GRCh38 accession
+        associated to ``tx_ac``.
 
-    def get_tx_exon_coords(
+        :param tx_ac: The RefSeq transcript accession to get exon data for.
+        :param genomic_ac: The RefSeq genomic accession to get exon data for.
+        :return: List of all exon coordinate data for ``tx_ac`` and ``genomic_ac``.
+            The exon coordinate data will include the exon number, transcript and
+            genomic positions for the start and end of the exon, and strand.
```

Comment:
Could we make a note that these aren't guaranteed to be sorted [even though I think they probably ~always are in practice?)

Reply:
Added order by clause

Reply:
mm that works too